File: ruby/weightconversion/WeightConversion.rb
#A ruby program for weight conversions
puts "A ruby program for weight conversions:"
loop do
  puts "--- Weight Conversion ---"
  puts "1. Kilogram -> Gram = (kilogram_value * 1000) grams"
  puts "2. Gram -> Kilogram = (gram_value / 1000) kilograms"
  puts "3. Pound -> Ounce = (pound_value * 16) oz"
  puts "4. Ounce -> Pound = (ounce_value / 16) lb"
  puts "5. Kilogram -> Pound = (kilogram_input * 2.205) lb"
  puts "6. Pound -> Kilogram = (pound_input / 2.205) kg"
  puts "7. Exit"
  print "Which operation you want to perform: "
  
  operation = gets.to_i
    # get option from user and convert into number
  break if operation ==7
  
  case operation
  when 1
    # Kilogram -> Gram = (kilogram_value * 1000) grams
    puts "What is the weight in kilograms?"
    kgweight = gets.to_i
    gramweight = kgweight* 1000
    puts "Converting the weight from kilogram in gram"
    puts "#{gramweight} gram"

  when 2
    # Gram -> Kilogram = (gram_value / 1000) kilograms
    puts "What is the weight in grams?"
    gramweight = gets.to_i
    kgweight = gramweight.to_f / 1000
    puts "Converting the weight from gram in kilogram"
    puts "#{kgweight} kg"
 when 3
    # Pound -> Ounce = (pound_value * 16) oz
    puts "What is the weight in Pounds?"
    poundweight = gets.to_i
    ounceweight = poundweight* 16
    puts "Converting the weight from Pound  in Ounce"
    puts "#{ounceweight} oz"
 when 4
    # Ounce -> Pound = (ounce_value / 16) lb
    puts "What is the weight in ounce?"
    ounceweight = gets.to_i
    poundweight = ounceweight.to_f / 16
    puts "Converting the weight from Ounce in Pounds"
    puts "#{poundweight} lbs"
 when 5
    # Kilogram -> Pound = (kilogram_input * 2.205) lb
    puts "What is the weight in kilograms?"
    kgweight = gets.to_i
    poundweight = kgweight * 2.205
    puts "Converting the weight from kilogram in Pounds"
    puts "#{poundweight} lbs"
 when 6
   # Pound -> Kilogram = (pound_input / 2.205) kg
    puts "What is the weight in Pounds?"
    poundweight = gets.to_i
    kgweight = poundweight.to_f / 2.205
    puts "Converting the weight from Pound in kilogram"
    puts "#{kgweight} kg"

  else
    puts "You have not typed a valid number, please run the program again."
      
  end
   puts ""   # needed to print next iteration in next line
end

File: python/weightconversion/WeightConversion.py

def convert():
    operation = input('''
Please type in weight conversion operation you would like to complete:
1 for Kilogram -> Gram = (kilogram_value * 1000) grams
2 for Gram -> Kilogram = (gram_value / 1000) kilograms
3 for Pound -> Ounce = (pound_value * 16) oz
4 for Ounce -> Pound = (ounce_value / 16) lb
5 for Kilogram -> Pound = (kilogram_input * 2.205) lb
6 for Pound -> Kilogram = (pound_input / 2.205) kg
''')

# Kilogram -> Gram = (kilogram_value * 1000) grams

    if operation == '1':
        kgweight = input("What is the weight in kilograms?")
        gramweight = (float(kgweight) * 1000)
        print('Converting the weight from kilogram in gram')
        print(f"{gramweight} gram ")

# Gram -> Kilogram = (gram_value / 1000) kilograms
    elif operation == '2':
          gramweight = input("What is the weight in grams?")
          kgweight = (float(gramweight) / 1000)
          print('Converting the weight from gram in kilogram')
          print(f"{kgweight} kg ")

# Pound -> Ounce = (pound_value * 16) oz
    elif operation == '3':
          poundweight = input("What is the weight in Pounds?")
          ounceweight = (float(poundweight) / 1000)
          print('Converting the weight from Pound  in Ounce')
          print(f"{ounceweight} oz ")

# Ounce -> Pound = (ounce_value / 16) lb
    elif operation == '4':
          ounceweight = input("What is the weight in ounce?")
          poundweight = (float(ounceweight) / 16)
          print('Converting the weight from Ounce in Pounds')
          print(f"{poundweight} lbs ")

# Kilogram -> Pound = (kilogram_input * 2.205) lb
    elif operation == '5':
          kgweight = input("What is the weight in kilograms?")
          poundweight = (float(kgweight) * 2.205)
          print('Converting the weight from kilogram in Pounds')
          print(f"{poundweight} lbs ")
# Pound -> Kilogram = (pound_input / 2.205) kg
    elif operation == '6':
          poundweight = input("What is the weight in Pounds?")
          kgweight = (float(poundweight) / 2.2046)
          print('Converting the weight from Pound in kilogram')
          print(f"{kgweight} kg ")

    else:
        print('You have not typed a valid number, please run the program again.')      
 
# Add again() function to calculate() function
    again()
def again():
    convert_again = input('''
Do you want to convert again?
Please type Y for YES or N for NO.
''')    
    if convert_again.upper() == 'Y':
        convert()
    elif convert_again.upper() == 'N':
        print('Thank You ! See you later.')
    else:
        again()

convert()